Project quotas aren't working after helm chart updating

Hi,

Under web ui the "Project quotas" shows "internal server error" after updating helm chart (https://helm.goharbor.io) from 1.4.2 to 1.5.0(current)
Project list isn't shown.

Environment:
OS: OEL 7.8
Harbor: v2.1.0
Docker version: 19.3.12
K8s version: 1.18.6

My yaml configuration:

---
  harborAdminPassword: "xxxxxxx"
  secretKey: "xxxxxxxxxxxxxxxxxxxxxx"
  logLevel: "info"
  expose: 
    type: "ingress"
    tls:
      enabled: true
      secretName: core-harbor-ingress
    ingress: 
      hosts: 
        core: "core-xxxxxxxx.com"
        notary: "notary-xxxxxxxx.com
      controller: default
      annotations:
        certmanager.k8s.io/cluster-issuer: vault-issuer
  externalURL: https://xxxxxxxx.com
  persistence:
    enabled: true
    resourcePolicy: "keep"
    persistentVolumeClaim: 
      registry: 
        existingClaim: "pvc-harbor"
        storageClass: "rook-cephfs"
        accessMode: ReadWriteMany
        size: 600Gi
      chartmuseum: 
        existingClaim: "pvc-harbor-chartmuseum"
        storageClass: "rook-cephfs"
        accessMode: ReadWriteMany
        size: 10Gi
      jobservice: 
        existingClaim: "pvc-harbor-jobservice"
        storageClass: "rook-cephfs"
        size: 4Gi
        accessMode: ReadWriteMany
      redis: 
        existingClaim: "pvc-harbor-redis"
        storageClass: "rook-cephfs"
        size: 8Gi
        accessMode: ReadWriteMany
    imageChartStorage: 
      type: "filesystem"
      filesystem:
        rootdirectory: /storage
      disableredirect: false
  jobservice: 
    jobLogger: "filesystem"
  database: 
    type: "external"
    external: 
      host: "pg-ei-registry-service"
      username: "xxxxxxxxxxxxxx"
      password: "xxxxxxxxxxxxxxxxxxxxxxxxx"
      coreDatabase: "registry"
      clairDatabase: "clair"
      notaryServerDatabase: "notary_server"
      notarySignerDatabase: "notary_signer"
      sslmode: "disable"
      port: "5432"
  redis: 
    type: "internal"
    external: 
      host: ""
      port: "6379"
      password: ""
  clair: 
    enabled: true
  notary: 
    enabled: true
  chartmuseum: 
    enabled: true
  cert: 
    enabled: true

`[harbor-harbor-core-6964bbb9cc-b92pn] 2020-10-30T18:30:57Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"INTERNAL_SERVER_ERROR","message":"failed to query database for quotas, error: project 4 not found"}]}  
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/lib/http/error.go:82, github.com/goharbor/harbor/src/lib/http.apiError 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/lib/http/error.go:51, github.com/goharbor/harbor/src/lib/http.SendError 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/common/api/base.go:67, github.com/goharbor/harbor/src/common/api.(*BaseAPI).RenderError 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/common/api/base.go:228, github.com/goharbor/harbor/src/common/api.(*BaseAPI).SendInternalServerError 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/core/api/quota.go:131, github.com/goharbor/harbor/src/core/api.(*QuotaAPI).List 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/reflect/value.go:460, reflect.Value.call 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/reflect/value.go:321, reflect.Value.Call 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/vendor/github.com/astaxie/beego/router.go:853, github.com/astaxie/beego.(*ControllerRegister).ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/middleware.go:52, github.com/goharbor/harbor/src/server/middleware.New.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/security/security.go:63, github.com/goharbor/harbor/src/server/middleware/security.Middleware.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/artifactinfo/artifact_info.go:55, github.com/goharbor/harbor/src/server/middleware/artifactinfo.Middleware.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/transaction/transaction.go:62, github.com/goharbor/harbor/src/server/middleware/transaction.Middleware.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/lib/orm/orm.go:73, github.com/goharbor/harbor/src/lib/orm.WithTransaction.func1 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/transaction/transaction.go:71, github.com/goharbor/harbor/src/server/middleware/transaction.Middleware.func1 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/notification/notification.go:31, github.com/goharbor/harbor/src/server/middleware/notification.Middleware.func1 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/orm/orm.go:53, github.com/goharbor/harbor/src/server/middleware/orm.MiddlewareWithConfig.func1 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/csrf/csrf.go:48, github.com/goharbor/harbor/src/server/middleware/csrf.attach.func1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/vendor/github.com/gorilla/csrf/csrf.go:297, github.com/gorilla/csrf.(*csrf).ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/csrf/csrf.go:68, github.com/goharbor/harbor/src/server/middleware/csrf.Middleware.func2 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/session/session.go:34, github.com/goharbor/harbor/src/server/middleware/session.Middleware.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/log/log.go:33, github.com/goharbor/harbor/src/server/middleware/log.Middleware.func1 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/requestid/requestid.go:37, github.com/goharbor/harbor/src/server/middleware/requestid.Middleware.func1 
[harbor-harbor-core-6964bbb9cc-b92pn] /harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:2836, net/http.serverHandler.ServeHTTP 
[harbor-harbor-core-6964bbb9cc-b92pn] /usr/local/go/src/net/http/server.go:1924, net/http.(*conn).serve `

--- a/make/migrations/postgresql/0050_2.2.0_schema.up.sql
+++ b/make/migrations/postgresql/0050_2.2.0_schema.up.sql
@@ -5,6 +5,13 @@ Fixes issue https://github.com/goharbor/harbor/issues/13317
 UPDATE role SET role_id=4 WHERE name='maintainer' AND role_id!=4;
 UPDATE role SET role_id=5 WHERE name='limitedGuest' AND role_id!=5;
 
+/*
+Clean the dirty data in quota/quota_usage
+  Remove quota/quota_usage when the referenced project not exists
+*/
+DELETE FROM quota WHERE reference='project' AND reference_id::integer NOT IN (SELECT project_id FROM project WHERE deleted=FALSE);
+DELETE FROM quota_usage WHERE reference='project' AND reference_id::integer NOT IN (SELECT project_id FROM project WHERE deleted=FALSE);
+
 ALTER TABLE schedule ADD COLUMN IF NOT EXISTS cron_type varchar(64);
 ALTER TABLE robot ADD COLUMN IF NOT EXISTS secret varchar(2048);
 

--- a/src/controller/quota/controller.go
+++ b/src/controller/quota/controller.go
@@ -132,10 +132,11 @@ func (c *controller) assembleQuota(ctx context.Context, q *quota.Quota, opts *Op
 
 		ref, err := driver.Load(ctx, q.ReferenceID)
 		if err != nil {
-			return nil, err
+			log.G(ctx).Warningf("failed to load referenced %s object %s for quota %d, error %v",
+				q.Reference, q.ReferenceID, q.ID, err)
+		} else {
+			q.Ref = ref
 		}
-
-		q.Ref = ref
 	}
 
 	return q, nil

